### PR TITLE
fix(plugin): fix plugin shutdown

### PR DESCRIPTION
**Pull request by zsoltkacsandi**
## Description

- introducing shutdown grace period in family manager to wait for the families to shut down properly in case of context cancellation (only needed for the plugin family)
- shutting down plugins if a panic occurs
- enabling init process for plugin containers to prevent zombie processes
- plugin family runner has been reworked to shut down the plugin container *before* it sends back the results
- passing context to Stop and Remove functions *without* cancel (otherwise in case of a context cancellation, these would be cancelled as well and wouldn't run)

## Type of Change

- [X] Bug Fix
- [ ] New Feature
- [ ] Breaking Change
- [ ] Refactor
- [ ] Documentation
- [ ] Other (please describe)

## Checklist

- [x] I have read the [contributing guidelines](https://github.com/openclarity/vmclarity/blob/main/CONTRIBUTING.md)
- [ ] Existing issues have been referenced (where applicable)
- [ ] I have verified this change is not present in other open pull requests
- [ ] Functionality is documented
- [ ] All code style checks pass
- [ ] New code contribution is covered by automated tests
- [ ] All new and existing tests pass

### CODE REVIEW

**Review by ramizpolic — DISMISSED** [review]

File: plugins/runner/internal/runtimehandler/docker/handler.go [hunk 1]
```diff
@@ -107,6 +108,7 @@ func (h *containerRuntimeHandler) Start(ctx context.Context) error {
 				},
 			},
 			Mounts: []mount.Mount{*scanDirMount},
+			Init:   &init,
```
Comment (ramizpolic) [inline plugins/runner/internal/runtimehandler/docker/handler.go:111]: we can do `to.Ptr(true)`

File: scanner/families/plugins/runner/runner.go [hunk 2]
```diff
@@ -67,77 +69,113 @@ func (s *Scanner) Scan(ctx context.Context, inputType common.InputType, userInpu
...
+		resChan <- res
+	}(ctx)
+
+	select {
+	case <-ctx.Done():
+		shutdownRunner(ctx)
+		return nil, errors.New("plugin context cancelled")
+	case r := <-resChan:
+		shutdownRunner(ctx)
+		return &types.ScannerResult{
```
Comment (ramizpolic) [inline scanner/families/plugins/runner/runner.go:174]: we can receive error here but it is not handled, we always assume the received result is valid

**Review by ramizpolic — CHANGES_REQUESTED** [review]

File: scanner/families/plugins/runner/runner.go (hunk 2 above)
Comment (zsoltkacsandi) [reply scanner/families/plugins/runner/runner.go:174]: Not exactly sure that I understand. Should I return an empty `ScannerResult` struct here?

`return &types.ScannerResult{}, errors.New("plugin context cancelled")`

Comment (ramizpolic) [reply scanner/families/plugins/runner/runner.go:174]: resChan result contains error which is not checked

Comment (zsoltkacsandi) [reply scanner/families/plugins/runner/runner.go:179]: Ah, thanks! Fixed.

**Review by ramizpolic — APPROVED** [review]

**Pull request merged**